src/Workspaces/Core/Portable/Formatting/Engine/TokenStream.cs

```diff
@@ -525,5 +560,17 @@ private int GetTokenIndexInStream(SyntaxToken token)
                 return new Iterator(_tokens);
             }
         }
+
+        private sealed class TokenOrderComparer : IComparer<SyntaxToken>
+        {
+            public static readonly TokenOrderComparer Instance = new TokenOrderComparer();
+
+            private TokenOrderComparer() { }
+
+            public int Compare(SyntaxToken x, SyntaxToken y)
+            {
+                return x.FullSpan.CompareTo(y.FullSpan);
```

Comment:
> Tokens have a .SpanStart that can be used that is cheap.

Yes, but it's not as cheap as `FullSpan`. If `SyntaxToken.Position` was public, it would be a totally different story.

Reply:
Wow.  You're right.  For some reason i thought SpanStart was a direct return with no computation.  We should really consider exposing Position (with the name 'FullSpanStart').  I would be 👍 on that :)